Replace Deploy/Retract with Flip

Geotricity opened the pull request:
- IntakeDeploy and IntakeRetract commands deleted
- IntakeFlip command added. This deploys the intake if it is retracted and retracts it if it is deployed

This makes it so that the deploy state of the intake can be controlled with 1 button instead of 2, in accordance with the button-command mapping on Discord.

Geotricity committed:
Replace Deploy/Retract with Flip

- IntakeDeploy and IntakeRetract commands deleted
- IntakeFlip command added. This deploys the intake if it is retracted and retracts it if it is deployed

This makes it so that the deploy state of the intake can be controlled with 1 button instead of 2, in accordance with the button-command mapping on Discord.

bricktrooper commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -42,6 +42,16 @@ public Intake()
+  retracted and retracts the intake if 
+  it is deployed
+  ===================================== */
+  public void flip() {
+    if (deployed) retract();
+    else deploy();
+  }
```
It's always better to put in brackets in case you need to add more lines of code later:
```
if (condition)
{
    // whatever
}
else
{
   // whatever
}
```
Or you could do this depending on your style:
```
if (condition) {
    // whatever
} else {
   // whatever
}
```

bricktrooper commented on src/main/java/frc/robot/commands/intake/IntakeFlip.java:
```diff
@@ -1,21 +1,20 @@
   private Robot robot;
 
-  public IntakeDeploy(Robot robot)
+  public IntakeFlip(Robot robot)
```
We should still keep the `IntakeDeploy` and `IntakeRetract` commands as well as `IntakeFlip`.  This will be useful in autonomous, where you may only want to deploy or retract the intake at certain points in time.  There won't be anyone using the controllers to toggle the intake.

Geotricity replied:
Good idea.

bricktrooper commented on src/main/java/frc/robot/commands/intake/IntakeFlip.java:
```diff
@@ -46,6 +45,6 @@ public void end(boolean interrupted)
   @Override
   public boolean isFinished()
   {
-    return true;
+    return false;
```
Why do we set this to `false`?  Doesn't that mean that the command will never exit?  We want this command to run once, then exit, every time it is triggered by the operator.

bricktrooper replied:
There's one exception where you might want to use `false`, and that would be a default command such as `Drive` that needs to run forever.  If another command requires the drivetrain, it will just interrupt `Drive`, go do the other command, and then come back and keep running `Drive`.

bricktrooper replied:
Basically if you want a command to be running forever in the background by default you can use `false`.  Otherwise if it's more of a one-shot thing you can use `true`.

Geotricity replied:
Thanks, that was a dumb mistake on my part.

bricktrooper reviewed (commented):
I agree with this change.  From an operator perspective, it makes sense to have one button that toggles the intake up and down.  But we should retain the original commands as well for use in autonomous and testing.

There is one thing to note here:  Just because the code thinks that the intake is deployed doesn't actually mean that it is.  For example, there might not be enough air pressure in the piston to deploy or retract in intake, so the value of `deployed` may not always be correct, since the code thinks it's deployed but mechanically it is not.  There really isn't any way to fix this without special hardware that can sense the piston position.  That's why it might be useful to keep explicit commands for deploy and retract.  But from an operator perspective, they can just press the `IntakeFlip` button again to re-sync the intake state, so it shouldn't be a huge issue.  And eventually the air will compress and the intake will move to the correct position.  Just something to think about.

For reference, the reason @droid-101 added the `deployed` variable was so to prevent the operator from spinning the intake when it is retracted.  So nothing super critical as of now.  `deployed` is always consistent with the position of the solenoid so we should be good.

Geotricity committed:
Add back Deploy and Retract

- Add back the Deploy and Retract Methods
- Change false to true in IntakeFlip.isFinished()
- Format if statement in Intake.flip()

bricktrooper reviewed (approved):
Looks good!  Should be good to merge.

bricktrooper commented:
Also @Geotricity when you have time, can you set the intake to be initially retracted inside the robot `initialize()` method?  Just so the intake doesn't accidentally deploy when the robot starts up.

Geotricity commented:
> Also @Geotricity when you have time, can you set the intake to be initially retracted inside the robot `initialize()` method? Just so the intake doesn't accidentally deploy when the robot starts up.

Yes, I can do that.
